components/button/button-styles.js

```diff
@@ -21,7 +21,10 @@ export const buttonStyles = css`
 		white-space: nowrap;
 		width: auto;
 	}
-	button:focus {
+	/* Outline on focus for elements not using FocusVisiblePolyfillMixin
+	 * or when focus-visible for those who are */
+	:host(:not([data-js-focus-visible])) button:focus,
+	button.focus-visible {
```

Comment:
Normally you'd just need `button.focus-visible { /* styles */ }`, but in this case a few things outside of core are using `buttonStyles` and aren't currently using the `FocusVisiblePolyfillMixin` -- with the first line here we're still applying the outline on normal focus if the host doesn't have a data attribute that the polyfill adds.